### DISCUSSION TITLE
better fee control

### DISCUSSION THREAD

**Discussion by viaj3ro** [ACINQ/eclair#2069] [Ideas]
`eclair.on-chain-fees.min-feerate` should be sats per kilobyte, like the other settings, not sats per byte to allow for better and more efficient fee control.

Same goes for channel closing and opening.

P.s.: it's quite annoying that I'm forced to select a category before posting which in turn deletes everything I wrote... Is Github to blame? Can it be fixed?

**Comment by t-bast** [comment]
We're using sat/byte almost everywhere because it's what maps best to what bitcoind offers.
How would using sats per kilobyte allow more efficient fee control for you?
You cannot go below 1 sat/byte anyway so I don't see how it's useful.

**Reply by viaj3ro** [reply]
sat/kb offers MUCH more precise fee control (1000x to be exact)! You can pay 1001 or 1100 s/kb and end up with 1.001 or 1.1 s/b. The latter tends to be the most price efficient. Whereas with sats/b you are stuck with full factor increments (as long as fractions are not supported as is right now).

bitcoinqt uses sats/kb, so sticking to bitcoind would actually mean s/kb or s/kw imho.

I'm fine with s/b though, as long as I can go down to .1 (with 1.1 being the lowest reasonable value obviously).

Increasing the factor by a full amount is quite a bad long-term strategy anyway the more valuable satoshis become over time.

**Reply by t-bast** [reply]
Ok, it is indeed more granular, so why not.

But I think if we change that, we'll wait for bitcoind to finally move to sat/kw everywhere, they are pushing to harmonize their units, so we won't do anything before they're done on their side and we can use the "best" unit they provide in the long term.
